packages/devtools_app/test/shared/diagnostics/inspector_service_test.dart

```diff
@@ -264,92 +264,131 @@ void main() {
+              equalsGoldenIgnoringHashCodes(
+                'inspector_service_tree_summary.txt',
+              ),
+            ),
+          );
+          final nodeInSummaryTree = findNodeMatching(root, 'MaterialApp')!;
+          expect(nodeInSummaryTree, isNotNull);
 
-        nodeInSummaryTree = findNodeMatching(root, 'Text')!;
-        expect(nodeInSummaryTree, isNotNull);
-        expect(
-          treeToDebugString(nodeInSummaryTree),
-          equalsIgnoringHashCodes(
-            'Text\n',
-          ),
-        );
+          expect(
+            treeToDebugString(nodeInSummaryTree),
+            equals(
+              equalsGoldenIgnoringHashCodes(
+                'inspector_service_node_summary.txt',
+              ),
+            ),
+          );
 
-        nodeInDetailsTree = (await group.getDetailsSubtree(nodeInSummaryTree))!;
+          await group.dispose();
+        });
 
-        expect(
-          treeToDebugString(nodeInDetailsTree),
-          equalsGoldenIgnoringHashCodes(
-            'inspector_service_text_details_tree.txt',
-          ),
-        );
+        test('isSummaryTree = false', () async {
+          await env.setupEnvironment();
+          final group = inspectorService!.createObjectGroup('test-group');
+          final RemoteDiagnosticsNode root = (await group.getRoot(
+            FlutterTreeType.widget,
+            isSummaryTree: false,
+          ))!;
+          // Tree contains all widgets.
+          expect(
+            treeToDebugString(root),
+            equals(
+              equalsGoldenIgnoringHashCodes(
+                'inspector_service_tree_no_summary.txt',
+              ),
+            ),
+          );
+          final nodeInTree = findNodeMatching(root, 'MaterialApp')!;
+          expect(nodeInTree, isNotNull);
+          expect(
+            treeToDebugString(nodeInTree),
+            equals(
+              equalsGoldenIgnoringHashCodes(
+                'inspector_service_node_no_summary.txt',
+              ),
+            ),
+          );
 
-        expect(nodeInDetailsTree.valueRef, equals(nodeInSummaryTree.valueRef));
-
-        await group.setSelectionInspector(nodeInDetailsTree.valueRef, true);
-        final selection = (await group.getSelection(
-          null,
-          FlutterTreeType.widget,
-          isSummaryTree: false,
-        ))!;
-        expect(selection, isNotNull);
-        expect(selection.valueRef, equals(nodeInDetailsTree.valueRef));
-        expect(
-          treeToDebugString(selection),
-          equalsIgnoringHashCodes(
-            'Text\n'
-            ' └─RichText\n',
-          ),
-        );
+          await group.dispose();
+        });
+
+        test('details tree', () async {
+          await env.setupEnvironment();
+          final group = inspectorService!.createObjectGroup('test-group');
+          final RemoteDiagnosticsNode root = (await group.getRoot(
+            FlutterTreeType.widget,
+            isSummaryTree: true,
```

Comment:
should this be false to test the details tree? If not can you add a comment to explain why?

Reply:
Added some comments :) This test is first getting a node in the summary tree, then using it to get the details tree for that node.